poc_mobile_verifier/src/speedtests.rs

```diff
@@ -163,6 +164,28 @@ impl SpeedtestAverages {
         self.speedtests.get(pub_key).map(Average::from)
     }
 
+    pub fn without_lapsed(self) -> Self {
+        let mut speedtests = HashMap::new();
+
+        for (pubkey, window) in self.speedtests.into_iter() {
+            let mut contiguous_tests = VecDeque::new();
+            let mut window = window.iter().peekable();
+            while let Some(st) = window.next() {
+                match window.peek() {
+                    Some(next_st)
+                        if (st.timestamp - next_st.timestamp)
+                            > Duration::hours(SPEEDTEST_LAPSE) =>
+                    {
+                        break
+                    }
+                    _ => contiguous_tests.push_back(st.clone()),
```

Comment:
Catchalls shouldn't generally be used for finite-variant types. It costs little extra to _write_ `None`, but is a boon to _read_. In the worst case it can cause a newly added variant to be handled with a handler one didn't really intend. Not really a problem for `Option` per se, but it's a good habit to have.
```suggestion
                    None => contiguous_tests.push_back(st.clone()),
```

Reply:
I don't think it can just be `None` in this case.  Because the previous match contains a guard clause there will be some values of `Some(speedtest)` that fall through to this branch.

Reply:
in this case i'm trying to catch two separate cases; i need to push the speedtest into the accumulator when the result of `window.peek()` is `None` _and_ when it is `Some(next_st)` with the implicit `if DIFFERENCE_IN_TIMESTAMPS < ARBITRARY_Duration`; explicitly writing `None` in the match loses the majority of the speedtests i care about accumulating and writing a third match head would duplicate the code in the match arm. I also _can't_ unfortunately match on `None | Some(next_st)` b/c the compiler will complain about a variable not being able to bind in all cases

Reply:
> I don't think it can just be `None` in this case. Because the previous match contains a guard clause there will be some values of `Some(speedtest)` that fall through to this branch.

Right! I totally looked passed the guard because it was on the next line _and_ I'm still getting used to guards being `if` and not `when`.

Reply:
> in this case i'm trying to catch two separate cases; i need to push the speedtest into the accumulator when the result of `window.peek()` is `None` _and_ when it is `Some(next_st)` with the implicit `if DIFFERENCE_IN_TIMESTAMPS < ARBITRARY_Duration`; explicitly writing `None` in the match loses the majority of the speedtests i care about accumulating and writing a third match head would duplicate the code in the match arm. I also _can't_ unfortunately match on `None | Some(next_st)` b/c the compiler will complain about a variable not being able to bind in all cases

`None | Some(_)` works for me locally.

Reply:
Case in point - explicit pattern made it very obvious for the reader (me, ahem ...) that there're actually 3 cases here, not 2.